frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditLongText/EditLongText.tsx

```diff
@@ -168,13 +168,19 @@ export const EditLongText = ({ field }: EditLongTextProps): JSX.Element => {
             name="ValidationOptions.customVal"
             control={control}
             rules={customValValidationOptions}
-            render={({ field }) => (
+            render={({ field: { onChange, ...rest } }) => (
               <NumberInput
                 flex={1}
+                inputMode="numeric"
                 showSteppers={false}
-                {...field}
                 placeholder="Number of characters"
                 isDisabled={!watchedSelectedValidation}
+                onChange={(val) => {
+                  // Only allow numeric inputs and return it as a number
+                  const numericValue = parseInt(val.replace(/\D/g, ''))
+                  onChange(isNaN(numericValue) ? 0 : numericValue)
```

Comment:
should we just not return if the parsed `numericValue` is `NaN`? this way the input will not suddenly change to 0 when non numbers are entered.

Reply:
Non numbers can't be entered anymore, the only allowed characters in the field is 0 to 9. The NaN case only occurs when the input is empty (so you get `parseInt('')` which is where the NaN comes from), which naturally corresponds to 0.

Reply:
u right, but I think in `frontend/src/utils/fieldValidation.ts`, `customVal` is declared to be of type `number|null`. To be sure, I'll change the type declaration as well.